tree/tree/src/TTree.cxx

```diff
@@ -6833,19 +6833,65 @@ bool TTree::MemoryFull(Int_t nbytes)
    return true;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Function merging branch structure from an outside tree into the current one
+///
+/// If the same branch name already exists in "this", it's skipped, only new ones
+/// are copied.
+/// Entries are not copied, just branch name / type is cloned
+/// Branches marked as DoNotProcess are not merged
+/// @param tree the outside TTree whose branches will be copied into this tree
```

Comment:
```suggestion
/// If this tree had some entries already in other branches and the new tree incorporates a new branch,
/// when importing the branch, we backfill the branch value with default values until GetEntries is reached,
/// to prevent misalignments in the TTree structure between branches / entries.
/// @param tree the outside TTree whose branches will be copied into this tree
```